src/views/project/project/pages/ProjectDetailPage.vue

```diff
@@ -231,6 +225,22 @@ export default {
                 vm.$store.dispatch('resource/provider/load'),
             ]);
         })();
+
+        onMounted(async () => {
+            await getProject(state.projectId);
+        });
+
+        watch(() => state.projectId, (projectId) => {
+            if (projectId) {
+                getProject(projectId);
+            }
+        }, { immediate: true });
+
+        watch(() => singleItemTabState.activeTab, (activeTab) => {
+            const routeName = find(singleItemTabState.tabs, { name: activeTab })?.routeName;
+            vm.$router.replace({ name: routeName }).catch(() => {});
+        });
```

Comment:
아래 제안은, 위에 tab item에 `routeName` 이 아닌 `name` 에 라우트 이름을 그대로 준다는 것을 가정하고 작성했어요!

```suggestion
<p-tab @update:activeTab="onUpdateActiveTab"/>

...

        const onUpdateActiveTab = (activeTab) => {
          vm.$router.replace({ name: activeTab }).catch(() => {});
        }
```

or 

```suggestion
        watch(() => singleItemTabState.activeTab, (activeTab) => {
            vm.$router.replace({ name: activeTab }).catch(() => {});
        });
```

watch vs `@update:activeTab` 에 이벤트 바인딩.
항상 고민이라, 다른 분들도 같이 고민하고싶어서.. 둘다 작성해봤습니다. ㅎㅎ

Reply:
전 항상 watch로 하는 편인데, watch가 많아지면 좀...지저분해보여서! 이번엔 @update를 활용하도록 하겠습니다!

Reply:
저도 이 update 방식 좋은거같아요!! 잘 기억해둘게요~